inference-engine/src/transformations/src/transformations/rt_info/mark_precision_sensitive_subgraphs.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/rt_info/mark_precision_sensitive_subgraphs.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include "itt.hpp"
+
+using namespace std;
+
+OPENVINO_RTTI_DEFINITION(ngraph::pass::MarkPrecisionSensitiveSubgraphs, "MarkPrecisionSensitiveSubgraphs", 0);
+
+void visit_shape_path(const shared_ptr<ngraph::Node>& node, set<shared_ptr<ngraph::Node>>& visited) {
+    if (!node)
+        return;
+    visited.insert(node);
+    deque<shared_ptr<ngraph::Node>> nodes{ node };
+    while (!nodes.empty()) {
+        auto curr_node = nodes.front();
+        nodes.pop_front();
+        // Do not check if already visited
+        if (ov::is_type<ngraph::opset8::ShapeOf>(curr_node)) {
+            continue;
+        }
+        visited.insert(curr_node);
+        if (ov::is_type<ngraph::op::Constant>(curr_node)) {
+            auto& rt_info = curr_node->get_rt_info();
+            rt_info["DISABLE_FP16_COMPRESSION"] = std::make_shared<ngraph::VariantWrapper<std::string>>("");
+        } else {
+            for (auto& input_value : curr_node->input_values()) {
+                // continue searching
+                const auto& input_node = input_value.get_node_shared_ptr();
+                nodes.push_front(input_node);
+            }
+        }
+    }
+}
+
+bool ngraph::pass::MarkPrecisionSensitiveSubgraphs::run_on_function(std::shared_ptr<ov::Function> f) {
+    deque<shared_ptr<Node>> nodes;
+    set<shared_ptr<Node>> visited;
+    for (auto& r : f->get_results())
```

Comment:
const auto&

Reply:
That is the parent declaration of this function, cant change it.